spec/support/shared_validator_specs.rb

```diff
@@ -16,6 +16,31 @@ def alphanumeric_string(length)
   end
 end
 
+shared_examples "a property with data-types restriction" do
+  it "fails if a wrong value is supplied" do
+    wrong_values.each do |val|
+      validator = described_class.new(entity_stub(entity, property => val))
+      expect(validator).not_to be_valid
+      expect(validator.errors).to include(property)
+    end
+  end
+
+  it "validates if a correct value is supplied" do
+    correct_values.each do |val|
+      validator = described_class.new(entity_stub(entity, property => val))
+      expect(validator).to be_valid
+      expect(validator.errors).to be_empty
+    end
+  end
+end
+
+shared_examples "a property that mustn't be empty" do
+  it_behaves_like "a property with data-types restriction" do
+    let(:wrong_values) { [""] }
```

Comment:
I think we should also add `nil` here, because the not_empty rule also prohibits nil.